src/phongo_bson.c

```diff
@@ -982,34 +896,140 @@ bool php_phongo_bson_data_to_zval(const unsigned char* data, int data_len, zval*
 	return retval;
 }
 
-/* Converts a BSON value to a ZVAL. */
-bool php_phongo_bson_value_to_zval(const bson_value_t* value, zval* zv)
+/** Converts a BSON value to a ZVAL, returning BSON objects and arrays as
+ * standard PHP types instead of Document or PackedArray instances */
+bool phongo_bson_value_to_zval_legacy(const bson_value_t* value, zval* zv)
```

Comment:
If you expect that a `false` return value will be accompanied by an exception, please note that in the comment.